packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2646,16 +2648,23 @@ class InputDecoration {
   /// vertically adjacent to) the input field.
   final String? labelText;
 
-  /// The style to use for the [labelText] when the label is above (i.e.,
-  /// vertically adjacent to) the input field.
+  /// The style to use for the [labelText] when the label is on top of the
+  /// input field.
   ///
-  /// When the [labelText] is on top of the input field, the text uses the
-  /// [hintStyle] instead.
+  /// When the [labelText] is above (i.e., vertically adjacent to) the input
+  /// field, the text uses the [floatingLabelStyle] instead.
   ///
   /// If null, defaults to a value derived from the base [TextStyle] for the
   /// input field and the current [Theme].
   final TextStyle? labelStyle;
 
+  /// The style to use for the [labelText] when the label is above (i.e.,
+  /// vertically adjacent to) the input field.
+  ///
+  /// If null, defaults to a value derived from the base [TextStyle] for the
```

Comment:
Shouldn't we specify that [labelStyle] is the default value? That should be required for the sake of backwards compatibility.

Reply:
Correct, fixed.